platform/pulp/platform/models/README.md

```diff
@@ -0,0 +1,121 @@
+Introduction
+============
+
+The pulp 3 data modeling effort is not just a translation or porting effort but instead
+an effort to build pulp 3 using pulp 2.  Remodeling will include changes to leverage the
+relational capabilities of postgres but also to improve upon the pulp 2 model.
+
+When creating each pulp 3 model object, consider the following:
+
+- Understand what the pulp 2 model (collection) stores and how the information
+  is used by pulp.
+
+- Name the model/table appropriately.  A table is implicitly plural.  Naming a
+  table `repository` is more appropriate than `repositories`.
+
+- Name the model class appropriately.  Each model object is a row in the table
+  an not a collection.  Class names should be singular.  For example, `Repository`
+  is a more appropriate class name than `Repositories`.
+
+- Do not use multiple inheritance (mixins) unless there is no other choice. Although
+  they are convenient, they greatly diminish the clarity of the model and as with
+  multiple inheritance in general, can result in untended behavior.  With a little extra
+  effort, a proper class hierarchy is almost aways possible and will be better in the end.
+
+- Question all fields.  Both the existence, type and naming.  Field names beginning with `_` 
+  should not exist in pulp 3.  The primary key field name is `id` and already defined in the
+  `Model` base class.  All `display_name` fields need to be renamed to `name`.  Also, avoid
+  redundant scope by prefixing field names (or any attribute) with the name of the class.
+  For example: `Task.task_type`.
+
+- Question all methods (including @property). We **only** want those methods that encapsulate
+  significant complexity and are widely used.  Most of the methods on pulp 2 models will likely
+  not be necessary or wanted in the pulp 3 models.  Let's keep the model interface as clean
+  as possible.
+
+- Most string fields will be `TextField`.  When the field is required and indexed,
+  use: `TextField(db_index=True)`.  When required but **not** indexed, 
+  use: `TextField(blank=False, default=None)`.  This ensures integrity at the model and
+  database layer(s) and supports validation at the REST layer.
+
+- All fields containing *ISO-8601* strings must be converted to `DateTimeField`.
+
+- Think about *natural* keys.  For example, each repository has a unique name (instead of repo_id
+  like in pulp 2) that is known to users and is not the primary key (`id` is).  The `name` is
+  the *natural* key.  Don't forget the index. See: https://en.wikipedia.org/wiki/Natural_key
+
+- Define a `natrual_key()` method in each model.  This both documents the *natural* key and
+  will be used by the django serializer.
+  See: https://docs.djangoproject.com/en/1.10/topics/serialization/#serialization-of-natural-keys
```

Comment:
Lets verify which version of Django we are targeting to ensure docs are pointing to that version